### ISSUE TITLE
[Linux] Brackets depends on obsolete libgcrypt11 package which is no longer included by default

### ISSUE THREAD

**Issue by Romane-T** [adobe/brackets#10255]
Brackets not currently installable on Jessie. Brackets has dependency on libgcrypt11, which is not available in the Jessie repositories. Jessie repositories have libgcrypt20. Brackets can be installed if user locates a copy of libgcrypt11, which can be co-installed with libgryypt20, but considering Jessie in freeze to become stable, doing this this should be unnecessary. Removing libgcrypt11 from system removes Brackets at the same time.
Romane

**Comment by prksingh** [comment]
@Romane-T Hey, could you try to build the appshell on your system. 
The repo is located at: https://github.com/adobe/brackets-shell. 
The build instructions are at: https://github.com/adobe/brackets-shell/wiki/Building-Brackets-Shell
This may help us locate if the error is with the installer.

You may want to check out this PR: https://github.com/adobe/brackets-shell/pull/489 
This may fix your issue but its not ready for testing yet.

**Comment by Romane-T** [comment]
On 21/01/15 21:17, Prashant Kumar Singh wrote:
>
> @Romane-T <https://github.com/Romane-T> Hey, could you try to build 
> the appshell on your system.
> The repo is located at: https://github.com/adobe/brackets-shell.
> The build instructions are at: 
> https://github.com/adobe/brackets-shell/wiki/Building-Brackets-Shell
> This may help us locate if the error is with the installer.
>
> You may want to check out this PR: adobe/brackets-shell#489 
> <https://github.com/adobe/brackets-shell/pull/489>
> This may fix your issue but its not ready for testing yet.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adobe/brackets/issues/10255#issuecomment-70820929>.
>
Good morning

Many thanks for your reply.

Sorry, but looked at the material at the links given, and too much over 
my head. The build instructions, though they seemed sort of straight 
forward, still left me confused. The last link to the "PR" - well, 
first, I have utterly no idea what a "PR" is, and what was on that page 
made absolutely zero sense to me.

I am _not_ a programmer. All that I learned about programming was a bit 
more than a quarter a century ago. Everything has dramatically since 
then, and I have forgotten it all anyway. I use Brackets solely to write 
my HTML pages, and even then fairly simple.

I have no idea what the difference between libgcrypt11 and libgcrypt20 
is, and have no interest in finding out. That is a developer issue.

Perhaps I am being simplistic, but it strikes me as a case of Brackets 
declaring a dependency for libgcrypt11, while Debian have updated in 
Jessie to libgcrypt20. If I remove libgcrypt11, Brackets is removed - 
that's the nature of a dependency. Debian does not carry Brackets in its 
repositories, so the issue returns to Brackets keeping up to what the 
distribution does.

Reason and logic thus tells me that attempting to do the build you 
suggested is unlikely to fix the problem __UNLESS__ this updates the 
dependencies for libgcrypt. If it doesn't update that dependency, then 
the problem still exists, and attempting to remove libgcrypt11 will 
still remove Brackets. And if it is the fix to the problem, then I am 
certain that at least one of your developers who knows something more 
about building and deploying will be quite capable of testing it and 
applying it for the next release, which I would assume they are doing 
anyway.

And while it is not fixed, Jessie users, except the small minority who 
keep a copy of libgcrypt11 on hand, will  not be users of Brackets, 
noting that Jessie is soon to go stable (172 bugs left to squash).

With greetings

Romane

**Comment by prksingh** [comment]
I think libgcrypt11 is required for CEF(https://code.google.com/p/chromiumembedded/) that most brackets features run on. The link i had posted is a link to a pull request that upgrades to a more recent version of CEF (I think libcrypt20 will work with the new CEF)

Here is where you can download the build for linux: https://github.com/adobe/brackets/releases/tag/linux-cef-2171

I don't have access to a system with Debian so I can't be sure this build will fix your issue. There may be a few other small glitches as this build is still being tested. I will try to update the release with a more stable build once ready. Thanks!

**Comment by Romane-T** [comment]
On 23/01/15 21:44, Prashant Kumar Singh wrote:
>
> I think libgcrypt11 is required for 
> CEF(https://code.google.com/p/chromiumembedded/) that most brackets 
> features run on. The link i had posted is a link to a pull request 
> that upgrades to a more recent version of CEF (I think libcrypt20 will 
> work with the new CEF)
>
> Here is where you can download the build for linux: 
> https://github.com/adobe/brackets/releases/tag/linux-cef-2171
>
> I don't have access to a system with Debian so I can't be sure this 
> build will fix your issue. There may be a few other small glitches as 
> this build is still being tested. I will try to update the release 
> with a more stable build once ready. Thanks!
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adobe/brackets/issues/10255#issuecomment-71182554>.
>
Good morning

Now _this_ is something within my capacity, and my thanks for giving me 
the opportunity.

Your first paragraph, I gather that whatever a CEF is, the new version 
might work with libgcrypt20

Downloaded the build in the second link, and before installing it, 
purged the version already installed. Once purged, tested to see if 
removing libgcrypt11 would affect any other package. There were no other 
dependencies.

Installed the version just downloaded from the link in your email. The 
install went without a hitch - zero warnings or errors.

Opened the application to ensure that it ran fine. it did. Noticed (an 
aside to the topic of this thread) on closing that, though still needing 
two clicks on the x button, the first click greyed the entire 
application window, then on the second click the window closed as 
expected. Previously, the first click on the x button did not appear to 
do anything, so it would appear that some progress has been made to 
resolving this issue. Closing through the 'File' menu closed the 
application normally.

After closing Brackets, attempted to remove libgcrypt11. Unfortunately, 
removing libgcrypt11 will still remove Brackets with it, so this build 
(Release 1.2 experimental build 1.2.0-15628 (master 5dddaa1f7) build 
timestamp: Thu Jan 22 2015 03:59:17 GMT-0800.) still retains the issue.

I even tried a reboot, knowing with 99% certainty that it would make no 
difference. It didn't.

With warm greetings

Romane

**Comment by MarcelGerber** [comment]
Some clarification here:
PR is short for "Pull Request" and it's when somebody proposes a change to the source code. It's pretty GitHub-specific, and as GitHub's all about social coding, it's a huge feature over here.

CEF is the "Chromium Embedded Framework", which is basically a configurable and embeddable version of the Chrome/Chromium browser. It's used in Brackets to render the UI, as Brackets is mostly written in HTML, CSS, and JavaScript - just like your website.

**Comment by Romane-T** [comment]
Good morning

This is, for a complete non-programmer as myself (I don't count the HTML 
I do as programming), quite useful information. Not just for the 
explanation of the acronyms - I feel a little less like an alien now in 
a foreign environment - but for the clear fact, combining information in 
both your and Prashant's posts, that the issue itself is not Brackets so 
much as upstream. Till upstream provide an update that resolves the 
issue, Brackets has its hands tied.

This is the only issue with Brackets and Jessie (Debian 8) that can 
become a show-stopper. Not difficult to resolve if one knows where to 
look. Have just checked to make sure, and Debian stable (Wheezy), due to 
become old-stable when Jessie is declared stable, has libgcrypt11. It 
will remain old-stable till the successor to Jessie goes stable and 
Jessie becomes old-stable, which gives a fair window of time during 
which the Chromium Embedded Framework can do the catchup. Perhaps just a 
note in a relevant place for the unaware that this dependency exists for 
Brackets at this time, and how they can obtain it - just a random 
thought. Installing via dpkg is then a breeze.

In the interim, if you would like someone to provide any testing on 
future development versions of Brackets to assess if this issue has been 
resolved, feel free to ask. It will be my pleasure, even if it is only a 
very tiny contribution back to the very excellent Brackets.

With greetings

Romane

**Comment by peterflynn** [comment]
Note: the issue with the close button is tracked as #4611, and it is indeed expected to be slightly improved in the test build linked above

**Comment by peterflynn** [comment]
I think to fully fix this we need to update the dependencies on our .deb build script to match the dependencies of the new CEF.  @jasonsanjose said this is kind of a pain (manual process), but we've done it in the past for some earlier CEF updates, and it seems necessary this time.

**Comment by Romane-T** [comment]
Good morning

On 28/01/15 07:35, Peter Flynn wrote:
>
> I think to fully fix this we need to update the dependencies on our 
> .deb build script to match the dependencies of the new CEF. 
> @jasonsanjose <https://github.com/jasonsanjose> said this is kind of a 
> pain (manual process), but we've done it in the past for some earlier 
> CEF updates, and it seems necessary this time.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adobe/brackets/issues/10255#issuecomment-71732625>.
>
Am I right in assuming that it is a tedious process rather than a 
programming process? If not a programming process, for someone like 
myself who would need perhaps some (or likely more) hand-holding as he

a)  has no idea how to do a pull request and the associated return of 
changed data
b) no idea of what actually needs to be changed

but is willing to learn provided there is someone with patience who can 
answer sometimes (often?) silly questions (in other words, I can appear 
really dumb sometimes), be of any use? Think of me as noob from core to 
surface.

The doing will, in the final analysis, prove the best teacher.

Besides (Romane grinning), it would give me something to do every third 
day. rather than my current cycle of every second day with existing 
projects.

With greetings

Romane

**Comment by MarcelGerber** [comment]
The dependency originates from https://github.com/adobe/brackets-shell/blob/master/installer/linux/debian/control#L8.
Changing it to `libgcrypt11 (>= 1.4.5) | libgcrypt20` should work, but I'm not sure if CEF/Chromium already support `libgcrypt20`.

@Romane-T Could you just test if `apt-get install chromium` works for you? (Of course in Jessie and without `libgcrypt11` installed) If it doesn't work, you can be sure that Brackets currently can't do anything about this issue. If it works, we'd still have to test if CEF works with `libgcrypt20`.

**Comment by Romane-T** [comment]
Good morning

On 01/02/15 07:23, Marcel Gerber wrote:
>
> The dependency originates from 
> https://github.com/adobe/brackets-shell/blob/master/installer/linux/debian/control#L8.
> Changing it to |libgcrypt11 (>= 1.4.5) | libgcrypt20| should work, but 
> I'm not sure if CEF/Chromium already support |libgcrypt20|.
>
> @Romane-T <https://github.com/Romane-T> Could you just test if 
> |apt-get install chromium| works for you? (Of course in Jessie and 
> without |libgcrypt11| installed) If it doesn't work, you can be sure 
> that Brackets currently can't do anything about this issue. If it 
> works, we'd still have to test if CEF works with |libgcrypt20|.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adobe/brackets/issues/10255#issuecomment-72337321>.
>

My current machine(s) are all running Jessie, bar one Windows machine 
which is seldom used.

Began by purging libgcrypt11, which uninstalled Brackets as a consequence.

executed aptitude install chromium. The version installed is 
chromium_40.0.2214.91-1 and install without errors.

Executed dpkg -i Brackets.Release.1.1.64-bit.deb

Errors reported - missing libgcrypt11. Transcript follows:

root@medion:/home/romane/software/linux/debs/Brackets# dpkg -i 
Brackets.Release.1.1.64-bit.deb
Selecting previously unselected package brackets.
(Reading database ... 209202 files and directories currently installed.)
Preparing to unpack Brackets.Release.1.1.64-bit.deb ...
Unpacking brackets (1.1.0-15558) ...
dpkg: dependency problems prevent configuration of brackets:
  brackets depends on libgcrypt11 (>= 1.4.5).

dpkg: error processing package brackets (--install):
  dependency problems - leaving unconfigured
Processing triggers for hicolor-icon-theme (0.13-1) ...
Processing triggers for menu (2.1.47) ...
Errors were encountered while processing:
  brackets

Going by your comment (above), looks like we are stuck at this for a bit 
yet.

With greetings

Romane

**Comment by sarathms** [comment]
So I tried what @MarcelGerber suggested above. But instead of changing it in the source, I changed it inside the distributed .deb file (`Brackets.Release.1.2.64-bit.deb`) and packaged it again. Brackets started working normally, so far. Not sure which feature in Brackets can trigger a possible crash.

Here's what I did:
* Extract the .deb file manually from your file manager. It creates a folder named `Brackets.Release.1.2.64-bit` (Take a backup of the debfile to avoid overwriting when repackaged).
* Edit the control file in the extracted package.
```
$ nano Brackets.Release.1.2.64-bit/DEBIAN/control
```
* Look for `libgcrypt11 (>= 1.4.5)` and change it to `libgcrypt20 (>= 1.5)`, save and exit.
* Change the ownership of all files and folder to root (This is needed for deb packaging purposes).
```
$ sudo chown -R root:root Brackets.Release.1.2.64-bit/
```
* Create the deb file again using the `dpkg-deb` command
```
$ sudo dpkg-deb --build Brackets.Release.1.2.64-bit/
```
* Install again
```
$ sudo dpkg -i Brackets.Release.1.2.64-bit.deb
```

Is it good to send a PR with this? I know it should go to brackets-shell, are there any guidelines for debian packaging?

**Comment by Romane-T** [comment]
Good morning

On 05/02/15 17:54, Sarath wrote:
>
> So I tried what @MarcelGerber <https://github.com/MarcelGerber> 
> suggested above. But instead of changing it in the source, I changed 
> it inside the distributed .deb file 
> (|Brackets.Release.1.2.64-bit.deb|) and packaged it again. Brackets 
> started working normally, so far. Not sure which feature in Brackets 
> can trigger a possible crash.
>
> Here's what I did:
>
>   * Extract the .deb file manually from your file manager. It creates
>     a folder named |Brackets.Release.1.2.64-bit| (Take a backup of the
>     debfile to avoid overwriting when repackaged).
>   * Edit the control file in the extracted package.
>
> |$ nano Brackets.Release.1.2.64-bit/DEBIAN/control
> |
>
>   * Look for |libgcrypt11 (>= 1.4.5)| and change it to |libgcrypt20
>     (>= 1.5)|, save and exit.
>   * Change the ownership of all files and folder to root (This is
>     needed for deb packaging purposes).
>
> |$ sudo chown -R root:root Brackets.Release.1.2.64-bit/
> |
>
>   * Create the deb file again using the |dpkg-deb| command
>
> |$ sudo dpkg-deb --build Brackets.Release.1.2.64-bit/
> |
>
>   * Install again
>
> |$ sudo dpkg -i Brackets.Release.1.2.64-bit.deb
> |
>
> Is it good to send a PR with this? I know it should go to 
> brackets-shell, are there any guidelines for debian packaging?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adobe/brackets/issues/10255#issuecomment-73007571>.
>
My thanks to Sarath. Following much head-banging on the desktop, finally 
succeeded in rebuilding the version 1.1 .deb with Sarath's instructions, 
and some help from the man command to fill in some spaces. I can confirm 
that the .deb installs fine, no error messages, and that removal of 
libgcrypt11 went smoothly, leaving Brackets installed.

But....

Though Brackets installs fine without libgcrypt11, it does not run 
without libgcrypt11. As soon as installed libgcrypt11 again, ran 
perfectly again. Tested by multiple removals and installs of both 
Brackets and libgcrypt11, together and separately.

Saranth, when you installed the modified .deb file, did you ensure that 
libgcrypt11 was removed from the system? If it is still there, that may 
account for it running for you. If you have no trace of libgcrypt11, 
then what is different between your system and mine that allows Brackets 
to run? Are you running Debian Jessie, or another version or another distro?

With greetings

Romane

**Comment by sarathms** [comment]
@Romane-T,
>Saranth, when you installed the modified .deb file, did you ensure that libgcrypt11 was removed from the system? If it is still there, that may account for it running for you. If you have no trace of libgcrypt11, then what is different between your system and mine that allows Brackets to run? Are you running Debian Jessie, or another version or another distro? 

I'm running Debian Jessie a.k.a testing a.k.a 8.0. Yes, I had no trace of `libgcrypt11` on my system. Instead I have `libgcrypt20`. Currently that is the only version of the package available in jessie. Its not possible for me to install `libgcrypt11`. Try `apt-cache search libgcrypt` and tell me if you still see `libgcrypt11`. However, you may find `libgcrypt11-dev` which has nothing to do with brackets.

>Though Brackets installs fine without libgcrypt11, it does not run without libgcrypt11. 
What do you mean? Does it not start? Does it crash?

>As soon as installed libgcrypt11 again, ran perfectly again.
How did you install libgcrypt11 again? If you are still able to, that means your package list isn't updated. You can try doing a `apt-get clean` and then `apt-get update` and `apt-get upgrade` if you haven't done it in a few weeks now.

**Comment by Romane-T** [comment]
Good morning

On 06/02/15 01:21, Sarath wrote:
>
> @Romane-T <https://github.com/Romane-T>,
>
>     Saranth, when you installed the modified .deb file, did you ensure
>     that libgcrypt11 was removed from the system? If it is still
>     there, that may account for it running for you. If you have no
>     trace of libgcrypt11, then what is different between your system
>     and mine that allows Brackets to run? Are you running Debian
>     Jessie, or another version or another distro?
>
> I'm running Debian Jessie a.k.a testing a.k.a 8.0. Yes, I had no trace 
> of |libgcrypt11| on my system. Instead I have |libgcrypt20|. Currently 
> that is the only version of the package available in jessie. Its not 
> possible for me to install |libgcrypt11|. Try |apt-cache search 
> libgcrypt| and tell me if you still see |libgcrypt11|. However, you 
> may find |libgcrypt11-dev| which has nothing to do with brackets.
>
>     Though Brackets installs fine without libgcrypt11, it does not run
>     without libgcrypt11.
>     What do you mean? Does it not start? Does it crash?
>
>     As soon as installed libgcrypt11 again, ran perfectly again.
>     How did you install libgcrypt11 again? If you are still able to,
>     that means your package list isn't updated. You can try doing a
>     |apt-get clean| and then |apt-get update| and |apt-get upgrade| if
>     you haven't done it in a few weeks now.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adobe/brackets/issues/10255#issuecomment-73063411>.
>
@saranth

First, re libcrypt11. My repositories are fine. I update and install 
available updates every day. Am not installing libcrypt11 from 
repositories, but from last copy of libcrypt11 left on my system when 
Jessie went to libcrypt20 - I kept it aside. Same as I must use dpkg to 
install Brackets, I use dpkg to install libcrypt11. Without that, would 
have been without Brackets for quite some time. Plus, libcrypt11 can be 
downloaded separately from repositories for Wheezy, but I never went 
down that track. Having Brackets on my system is more important than 
having a pristine Jessie.

But none of that is important to the issue at hand, it is a mere 
sidetrack, though I thank you for your advice. What is important is that 
for you, Brackets runs fine on libcrypt20. For me, it does not start and 
i have to have libcrypt11 on board. Crash? No, not that I know of. Does 
not start. Start-up indicator, but that stops at the default 30 seconds, 
then nothing. There is likewise no trace showing in the System Monitor 
during or after the start-up indicator showing. All of that says that 
Brackets is recognised but not loading, but beyond that I have no 
knowledge, and in this circumstance have no expertise in diagnosing.

Which leaves us with the important question - what is different between 
our two systems? If we can work that out, then perhaps we can find a 
solution that works for everyone, not just a few. So, can you or anyone 
else tell me what I need to look at to try and resolve this system 
difference? What sort of things can I try?, hopefully to duplicate the 
experience of Saranth.

With greetings

Romane

**Comment by lrebrown** [comment]
>Which leaves us with the important question - what is different between our two systems?

@Romane, sarathms was repacking a preview of Brackets v1.2, while you stated earlier that you are repacking v1.1. There is the problem. You cannot expect v1.1 to just work the same like that.

In v1.1 the Brackets team updated the version of the CEF component Brackets is built upon. The primary reason for doing this was to address some issues with using Brackets in Linux. Unfortunately other issues cropped up in Brackets when using this new version, so while they were able to ship v1.1 for Windows and Mac with the new CEF, v1.1 for Linux retained the previous version. For v1.2 they have been working on resolving those issues to now allow them to provide us with a Linux build built upon this new version of CEF. I expect that CEF has a dependency on libcrypt, with the older version of CEF working with and being built against libcrypt11, and the newer CEF, libcrypt20. If that is so, you cannot just repackage the Linux build of Brackets v1.1 with a libcrypt20 dependency, because the older version of CEF it uses is not necessarily compatible with the new version of libcrypt. This should explain this issues you are experiencing.

Try instead to repackage the Linux preview build of v1.2 with libcrypt20, as done by sarathms.

Hopefully if it is the case that v1.2 works with and requires libcrypt20, the Brackets team will package it with this updated dependency when they release it, otherwise we'll have an issue, requiring us to repackage with correct dependency ourselves, which would obviously not be ideal.

**Comment by Romane-T** [comment]
Good morning

On 06/02/15 10:12, lrebrown wrote:
>
>     Which leaves us with the important question - what is different
>     between our two systems?
>
> @Romane <https://github.com/Romane>, sarathms was repacking a preview 
> of Brackets v1.2, while you stated earlier that you are repacking 
> v1.1. There is the problem. You cannot expect v1.1 to just work the 
> same like that.
>
> In v1.1 the Brackets team updated the version of the CEF component 
> Brackets is built upon. The primary reason for doing this was to 
> address some issues with using Brackets in Linux. Unfortunately other 
> issues cropped up in Brackets when using this new version, so while 
> they were able to ship v1.1 for Windows and Mac with the new CEF, v1.1 
> for Linux retained the previous version. For v1.2 they have been 
> working on resolving those issues to now allow them to provide us with 
> a Linux build built upon this new version of CEF. I expect that CEF 
> has a dependency on libcrypt, with the older version of CEF working 
> with and being built against libcrypt11, and the newer CEF, 
> libcrypt20. If that is so, you cannot just repackage the Linux build 
> of Brackets v1.1 with a libcrypt20 dependency, because the older 
> version of CEF it uses is not necessarily compatible with the new 
> version of libcrypt. This should explain this issues you are experiencing.
>
> Try instead to repackage the Linux preview build of v1.2 with 
> libcrypt20, as done by sarathms.
>
> Hopefully if it is the case that v1.2 works with and requires 
> libcrypt20, the Brackets team will package it with this updated 
> dependency when they release it, otherwise we'll have an issue, 
> requiring us to repackage with correct dependency ourselves, which 
> would obviously not be ideal.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adobe/brackets/issues/10255#issuecomment-73157609>.
>
I sit here now giggling with glee. You have, @Irebrown, hit the nail on 
the head. My faulty assumption, for which I paid the penalty.

Results achieved by Sarath now duplicated on my machine.

Just for the record:

Re-downloaded 1.2 from the link given earlier by Prashant on 23 Jan 15.
Rebuilt the .deb as per the directions from Sarath on 5 Feb 15 (didn't 
take me three hours this time).
Removed libgcrypt11, which automatically removed Brackets.
Attempted to install the downloaded .deb of Brackets 1.2, to confirm 
that would fail - no libgcrypt11. Confirmed.
Removed broken Brackets package.
installed rebuilt package - dpkg -i Brackets.Release.1.2.64-bit.deb - no 
errors or warnings
Clicked on icon for Brackets in menu's - worked perfectly; no 
hesitation, no stalling, no crashes.
Closed and restarted Brackets a number of times to confirm start-up 
resolved. Is clear from the post by Sarath when posting the fix that 
have now duplicated what is happening on (her? his?) computer.

With peoples permission, I would like to offer a commendation to those 
here for dealing with my sometimes silliness with patience and 
perseverance, and allowing my continued participation in the 
conversation (i.e. not ignoring me) and providing me with opportunities 
to contribute positively. I am no developer by any standard, but what I 
have learned from the people here has been useful and gratifying. Thank you.

Now, if there is any further testing I can do in this or other areas, I 
would be most happy to impose again on your patience and generosity 
(Romane grinning).

With greetings

Romane

**Comment by redmunds** [comment]
@Romane-T 

> Now, if there is any further testing I can do in this or other areas, I would be most happy to impose again on your patience and generosity (Romane grinning).

We will be posting Release 1.2 beta builds soon, so please help us test those. It will be announced on the [brackets-dev forum](https://groups.google.com/forum/?fromgroups#!forum/brackets-dev).

**Comment by jasonsanjose** [comment]
@MarcelGerber, this worked great:

>Changing it to libgcrypt11 (>= 1.4.5) | libgcrypt20 should work, but I'm not sure if CEF/Chromium already support libgcrypt20.

Thanks for the tip! I created a PR https://github.com/adobe/brackets-shell/pull/501.

**Comment by jasonsanjose** [comment]
@Romane-T @MarcelGerber I posted a new build here https://github.com/adobe/brackets/releases/download/linux-cef-2171/Brackets.Release.1.2.64-bit.deb that contains fix for this bug (see PR https://github.com/adobe/brackets-shell/pull/501) as well as including the CEF 2171 (see https://github.com/adobe/brackets-shell/pull/499).

I was able to install this on a clean (mostly) install of debian 8. I had installed the `build-essential` package early on. Upon install of Brackets, I was prompted to install `libcurl3 which installed without issues.

If you get some time to try this new build, please let us know how it goes.

**Comment by lrebrown** [comment]
@jasonsanjose 
> I posted a new build...

Played with it briefly in a Debian sid VM. It installed without issue (this VM has both libcrypt20 and libcrypt11 installed, the latter is still available and used by some things in sid). The window overlap issue is no more with the new CEF build - fantastic! Closing brackets both via the 'x' and via file > quit works. The only issue I've noticed is that the window doesn't remember the position or size I expand it to upon reopening - rather irritating.

I'm using gnome and I'm running this VM in virtualbox.

Oh, having said that, I just tried to close it via the 'x' (to see if it remembers position or not) and I'm left with an empty window which won't close :/ And now it's doing that every time. I'll mention it over in #499

**Comment by Romane-T** [comment]
On 11/02/15 04:59, Jason San Jose wrote:
>
> @Romane-T <https://github.com/Romane-T> @MarcelGerber 
> <https://github.com/MarcelGerber> I posted a new build here 
> https://github.com/adobe/brackets/releases/download/linux-cef-2171/Brackets.Release.1.2.64-bit.deb 
> that contains fix for this bug (see PR adobe/brackets-shell#501 
> <https://github.com/adobe/brackets-shell/pull/501>) as well as 
> including the CEF 2171 (see adobe/brackets-shell#499 
> <https://github.com/adobe/brackets-shell/pull/499>).
>
> I was able to install this on a clean (mostly) install of debian 8. I 
> had installed the |build-essential| package early on. Upon install of 
> Brackets, I was prompted to install `libcurl3 which installed without 
> issues.
>
> If you get some time to try this new build, please let us know how it 
> goes.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adobe/brackets/issues/10255#issuecomment-73760097>.
>
Good morning

System is Debian Jessie running KDE. This machine has been running 
'testing' since Lenny was in 'testing', though full re-install at one 
stage when I really messed something up while 'Wheezy' was in testing. I 
did once contemplate (re)learning C, and installed the relevant 
libraries, but do not think that these would have any effect.

For various purposes, had downgraded to Brackets version 1.0. I need the 
functionality in the extension 'right-click-extended', which every 
version of Brackets since version 1.0 has prevented from functioning - 
has been reported appropriately for the extension.

Purged Brackets and libgcrypt11
libcurl3 is already installed on machine.
installed this build of Brackets. No errors, no warnings.
Opened Brackets:
     1st test - close via 'x'. For the first time ever that have seen 
since using Brackets, Brackets closed with only a single click. Yaaaay - 
good step forward. Tested about five times, same each time.
     2nd test - close via 'File->Quit'. as with closing via 'x', 
immediate close. Again, tested about five times.

There was an update for one of the extensions available. Updated without 
any problems - no warnings, no errors.

I had Chromium installed from a previous test for Brackets. Removed 
Chromium. No change to above results.

Extension 'right-click-extended' still fails to work. Something changed, 
obviously, in the code-base since version 1.0 which directly affects 
this extension.

@Irebrown - regarding not remembering position, I have never known 
Brackets to remember position, and must resize at every opening. Only a 
minuscule irritation for self not affecting general satisfaction with 
Brackets, but still, would be nice :)

Will leave this build installed to test functionality in my usage needs. 
Should issues be reported to a certain thread, or simply reported in the 
same manner as the stable builds?

I have an old and slow 32 bit machine running Jessie with lxqt for the 
desktop environment, all freshly installed about two months ago. My out 
and about machine. Looking forward to a 32 bit build to test on this 
machine when such becomes available.

With greetings

Romane

**Comment by jasonsanjose** [comment]
@Romane-T Here's a 32-bit installer https://github.com/adobe/brackets/releases/download/linux-cef-2171/Brackets.Release.1.2.32-bit.deb.

I tested this on a clean Debian 7 install. Looks like `libcurl3` is the only missing package again.

@nethip On my 32-bit VM I'm running into the crash on quit more frequently.

**Comment by Romane-T** [comment]
Good morning

On 14/02/15 09:17, Jason San Jose wrote:
>
> @Romane-T <https://github.com/Romane-T> Here's a 32-bit installer 
> https://github.com/adobe/brackets/releases/download/linux-cef-2171/Brackets.Release.1.2.32-bit.deb.
>
> I tested this on a clean Debian 7 install. Looks like |libcurl3| is 
> the only missing package again.
>
> @nethip <https://github.com/nethip> On my 32-bit VM I'm running into 
> the crash on quit more frequently.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adobe/brackets/issues/10255#issuecomment-74344337>.
>
Became excited to see this. Many many thanks.

System is fully up-to-date Debian Jessie on a Samsung N120, with lxqt 
version 9.0-1 as the desktop environment. This was a completely fresh 
install at the time lxqt was in version 8 just a few months ago, and has 
been put together considering the age and speed of the machine. libcurl3 
is already installed in relation to other packages (dunno which ones, 
it's just there).

Before installing, purged both the previous version of Brackets, and 
also libgcrypt11.

First attempt at install (dpkg -i Brackets.Release.1.2.32-bit.deb) 
failed - missing libgcrypt11. As per comment from @Sarath on 5 February, 
re-packaged deb. This machine is so slow, that the build itself took 
about 10 minutes to complete :) :)

Second attempt at install successful - no errors, no warnings.

At start, Brackets presents a window almost instantly (for this computer 
- about three or so seconds), but the remainder slow due to the old 
hardware.

Tested closing, both by 'File->Quit' and by the 'x', multiple times. No 
crashes at any attempt. Either way, Brackets closed quickly and cleanly.

There were 6 updates to extensions available (binawhile). Updates 
installed and updated with no issues - no errors, no warnings.

Tomorrow will use this version of Brackets on the Samsung to do all my 
HTML, to give it a good run (today is .pdf day). There is an issue with 
the keyboard which have still to report for the 64 bit version 
downloaded earlier via this thread (will do today) so will be 
interesting to see if translates into this 32 bit version.

@Jason - just an aside, and hopefully not trying to teach grandma to 
suck eggs - thinking that the crash is to do with the VM rather than 
Brackets, as there are no crashes on the packages running on my systems 
(no VM), and @Sarath noted in his contribution on 5 February that after 
his repackaging, Brackets ran normally (but keywords are 'so far'). Have 
a memory of mention of crashing in a VM before.

With greetings

Romane

**Comment by lrebrown** [comment]
@Romane-T, how many CPU cores do your systems have? All crashes so far I believe have been only happening in VMs with a single cpu core, and not with VMs with multiple cores (not tested a multi core VM myself). I do not know if anyone has yet tested it directly on a machine with only one CPU core. Assuming it's VM related is jumping the gun here. :)

**Comment by Romane-T** [comment]
Good morning

On 14/02/15 13:33, lrebrown wrote:
>
> @Romane-T <https://github.com/Romane-T>, how many CPU cores do your 
> systems have? All crashes so far I believe have been only happening in 
> VMs with a single cpu core, and not with VMs with multiple cores (not 
> tested a multi core VM myself). I do not know if anyone has yet tested 
> it directly on a machine with only one CPU core. Assuming it's VM 
> related is jumping the gun here. :)
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adobe/brackets/issues/10255#issuecomment-74359634>.
>
Why I said about teaching grandma to suck eggs :) - that should tell 
people not to take too much notice of that paragraph.

My machine is a  Medion PC with four cores.

The Samsung has two cores.

Was hoping one of my other machines (quite old hardware) might be single 
core, but both two core. So puts paid to that plan for testing.

Is it still possible to get single-core machines? Will see if can find 
one that can test on.

With greetings

Romane

**Comment by MarcelGerber** [comment]
@Romane-T It's proably easier to test in a multi-core VM than using single-core hardware (even though that'd be helpful as well to confirm the suspicion).

**Comment by Romane-T** [comment]
Good morning

On 14/02/15 09:17, Jason San Jose wrote:
>
> @Romane-T <https://github.com/Romane-T> Here's a 32-bit installer 
> https://github.com/adobe/brackets/releases/download/linux-cef-2171/Brackets.Release.1.2.32-bit.deb.
>
> I tested this on a clean Debian 7 install. Looks like |libcurl3| is 
> the only missing package again.
>
> @nethip <https://github.com/nethip> On my 32-bit VM I'm running into 
> the crash on quit more frequently.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adobe/brackets/issues/10255#issuecomment-74344337>.
>
On my Samsung 32 bit machine, the keyboard fails completely - unable to 
type anything or move curser with keys. Mouse works, however. Reloaded 
without extensions, still the same. Will report via the usual 
bug-reporting channel.

With greetings

Romane

**Comment by jasonsanjose** [comment]
>First attempt at install (dpkg -i Brackets.Release.1.2.32-bit.deb)
failed - missing libgcrypt11. As per comment from @Sarath on 5 February,
re-packaged deb. This machine is so slow, that the build itself took
about 10 minutes to complete :) :)

Oops. Thanks, @Romane-T. I built the 32-bit build on a slightly different branch than the 64-bit one. It doesn't have the fix for `libgcrypt11`. 

It looks like I missed the train for the 1.2 release. Hopefully this will all be settled for 1.3.

**Comment by a-moses** [comment]
Hello @jasonsanjose @Romane-T @MarcelGerber

look on github's Atom simple solution for this : 

https://github.com/atom/atom/pull/5605

just change file.deb /DEBIAN/control  from:
' libgcrypt11 (>= 1.4.5) '  TO :    ' libgcrypt11 (>= 1.4.5) | libgcrypt20  '

*libgcrypt20 come with backward compatibility to libgcrypt11

**Comment by MarcelGerber** [comment]
@a-moses Yeah, that's what we have in mind and what's already in PR https://github.com/adobe/brackets-shell/pull/501.

**Comment by a-moses** [comment]
@MarcelGerber  Yeah letter i saw that ... so why not to do this like the original PR ?

**Comment by Romane-T** [comment]
Have just installed the latest version of Brackets available via brackets.io. Still has dependency on libgcrypt11. Tried modification as per @sarathms 29 days ago. Brackets fails to start - no crash, just failed to start, did not even show in the system monitor. As soon as re-installed libgrypt11 (without doing another install of either the modified Brackets or the downloaded Brackets), Brackets worked fine again.

**Comment by nethip** [comment]
@Romane-T Could you run Brackets from terminal and post your results? I am guessing this has to to with GLIB_C version. But it is just a guess waiting to be validated. From terminal execute the following commands.

```
cd  /opt/brackets/
./Brackets
```

**Comment by nethip** [comment]
@Romane-T We have not merged the PR for this release and that is why the problem is still present. Please expect this to be fixed in the next release. We will be actively looking at this issues.

**Comment by Romane-T** [comment]
Good morning Prasanth

On 03/03/15 16:54, Prashanth Nethi wrote:
>
> @Romane-T <https://github.com/Romane-T> We have not merged the PR for 
> this release and that is why the problem is still present. Please 
> expect this to be fixed in the next release. We will be actively 
> looking at this issues.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adobe/brackets/issues/10255#issuecomment-76896097>.
>
Ok, can live with this no problem. Main thing is that I have it 
functioning and usable.

It may be worth a note for Debian Jessie users that if they wish to 
install libgcrypt11, that is can be obtained from the (is it Wheezy, the 
previous release? I have forgotten already) repositories, whether by 
apt/aptitude/synaptic, or by manual download and install.

In response to your previous email, the following:

romane@medion:~$ cd /opt/brackets
romane@medion:/opt/brackets$ ./Brackets
./Brackets: error while loading shared libraries: libudev.so.0: cannot 
open shared object file: No such file or directory
romane@medion:/opt/brackets$

This is with libgcrypt11 installed, version 1.2 as available via 
brackets.io (with modification as mentioned in earlier posting). Runs 
fine and is stable, however, from the icon in the menu's.

Looking forward already to the next release :)

With greetings

Romane

**Comment by alorgill** [comment]
Issue still persists in Fedora 21 -- installed via Fedy.

`/usr/bin/brackets: error while loading shared libraries: libgcrypt.so.11: cannot open shared object file: No such file or directory`

**Comment by dimr** [comment]
hi,
in case anyone is looking for the libgcrypt11.deb file it is available here http://snapshot.debian.org/package/libgcrypt11/1.5.4-3/#libgcrypt11_1.5.4-3

**Comment by mihailik** [comment]
Is there a simple workaround then? Or a complex workaround for that matter?

Brackets still doesn't work for the latest Ubuntu Vivid (15).

It's clear some dependency is missing,

**Comment by mihailik** [comment]
Note that the standard procedure for the brackets-shell build fails too, with an error like 'pkg-config missing'.

Installing pkg-config, now libgtk2.0 is missing.

Installing that gives a genuine C++ compilation error something like appshell/appshell_extensions_platform.h:65:57: error: ‘GTK_WIDGET’ was not declared in this scope.

That's what I mean by Voodoo. Not just a minor glitch easily fixable by looking at the error, but some bazlion-thousand intertwined glitches with every next one more puzzling that the previous.

**Comment by luisalvarado** [comment]
I find this information for Debian/Ubuntu users http://www.webupd8.org/2015/04/fix-missing-libgcrypt11-causing-spotify.html

Will simply wait to install Brackets after it supports libgcrypt20. I thought I was the only one with the issue.

**Comment by gogobook** [comment]
Brackets can't install in ubuntu(15.04), it need libgcrypt11(>=1.4.5)

**Comment by luisalvarado** [comment]
Ok, do not know if it's Brackets 1.3 or Ubuntu 15.04 right now (Since 3 days ago at least). But installing Brackets on Ubuntu 15.04 works perfect. No problems. Tested this on 5 laptops and 2 PCs. All using Ubuntu 15.04 64-bit.

**Comment by gogobook** [comment]
My PC is Ubuntu 15.04 64-bit too, installing Brackets 1.2 is fine. But Brackets 1.3 is not.

**Comment by js0511** [comment]
NAVER - http://www.naver.com/
--------------------------------------------

oiup2006@naver.com 님께 보내신 메일 <Re: [brackets] Debian - libgcrypt version breaks Brackets. (#10255)> 이 다음과 같은 이유로 전송 실패했습니다.

--------------------------------------------

받는 사람이 회원님의 메일을 수신차단 하였습니다. 


--------------------------------------------

**Comment by nethip** [comment]
@gogobook You mean you are unable to install it or is it that you are not able to launch it? If you are not able to install, you could try using `dpkg` to run brackets installer from command line and see where the installer is failing. If it is about launching Brackets, again try launching Brackets from command line, something like typing `/opt/Brackets/Brackets` in command line where `/opt/Brackets` is the install location, and see what errors appear there.

**Comment by Romane-T** [comment]
Debian Jessie, no problems with 1.3. But...

I still have libgcrypt11 installed from when bug first notifies way back 
when. My first suggestion would be to make certain that your machine 
actually has all the required dependencies (i.e. libgcrypt11)

**Comment by gogobook** [comment]
Thank you, Romane-T and nethip. After manually install libgcrypt11. Brackets 1.3 can be installed. And it could be launched by command line.

**Comment by Damey37** [comment]
Hi guys ! It You must install the package "libgcrypt20" and "libgcrypt20-dev" first. Then, recompile the package .deb and changed in the file "control" the dependence "libgcrypt11" to "libgcrypt20". Finally, rebuilt a package .deb of folder "Brackets" . And it work !

**Comment by itsakt** [comment]
I just downloaded Brackets 1.3 and had same issue in Debian 8. I went here http://ftp.acc.umu.se/mirror/cdimage/snapshot/Debian/pool/main/libg/libgcrypt11/ 
and installed libgcrypt11 for my system and now Brackets installed successfully

**Comment by sarangbaheti** [comment]
thanks @itsakt for the location for libgcrypt11, installed 1.4.5 version and got things rolling..

**Comment by lucaciavatta** [comment]
Can't install Brackets 1.3 on Ubuntu 15.04 because 'Dependency is not satisfiable: libgcrypt11
 (>= 1.4.5). I followed the itsakt workaround and now Brackets is installed successfully.

**Comment by luisalvarado** [comment]
Did a fresh install of Ubuntu 15.04. Problem mention here is happening.

**Comment by sarangbaheti** [comment]
@luisalvarado try installing libgcrypt11 version 1.4.5 from http://ftp.acc.umu.se/mirror/cdimage/snapshot/Debian/pool/main/libg/libgcrypt11/  and it should work

**Comment by luisalvarado** [comment]
@sarangbaheti  Thank you. It did.

**Comment by NicolasKAROLAK** [comment]
Installing a package which is not in the repo is not a good solution. Software are upgraded for a good reason, like bugs and security issues.

The suggested workaround (extract .deb, modify dependancies et rebuild) doesn't work for me :

```
/usr/bin/brackets: error while loading shared libraries: libgcrypt.so.11: cannot open shared object file: No such file or directory
```

**Comment by efolio** [comment]
Same dependency issue here, on latest Ubuntu, fresh install.
This is not a low priority bug anymore. Please update the label.

**Comment by Spangenhelm** [comment]
:+1: @sarangbaheti !
Installing this package: [libgcrypt11_1.5.4-3_amd64.deb](http://ftp.acc.umu.se/mirror/cdimage/snapshot/Debian/pool/main/libg/libgcrypt11/libgcrypt11_1.5.4-3_amd64.deb) solved my problem of not beeing able to install Bracket 1.3 on Ubuntu Mate 64-bit

**Comment by abairo** [comment]
Thank you @sarangbaheti, It's worked for me.

**Comment by HeartBreakOne** [comment]
Same dependency issue, with a Ubuntu 15.04 64 bits
Installing libgcrypt11 through the .deb file solves the problem.

**Comment by dwilmer** [comment]
Yes, installing the older package makes this problem go away. No, I do not think requiring users to manually install an outdated package is a good solution. Furthermore: it's a *crypto* package, meaning that security is at stake when not using the most recent version.

It's been 7 months since this bug was reported, some sources mention a fix in version 1.3, but progress on an actual fix is not really reported. What is the status?

**Comment by Spangenhelm** [comment]
@dwilmer i agree this needs to be fixed !

**Comment by mihailik** [comment]
I say -- if they don't watch this thread, let's hijack it for something remotely useful.

My first question, what's the general opinion on blogging software/tools? I want to revive my old forgotten blog, but I've no idea what's the kosher way to do it?

All these PHP Wordpress and Ruby 'simple tools' look a bit clunky to my untrained eye. Or am I looking the wrong way?

**Comment by Spangenhelm** [comment]
Hi @mihailik i'm not sure that i agree with your proposition but i don't mind answering it: i suggest you to take a look at http://nibbleblog.com (a nifty blog/cms written in php without database) or, from the same creator: http://www.bludit.com (same kind, in beta for now, you can check the differences between the two projects here: http://blog.nibbleblog.com/post/bludit-my-new-project/)
Cheers +++

**Comment by mackenza** [comment]
@mihailik I would be happy to discuss this on Slack or in the Google Groups... but hijacking this issue is not cool.

**Comment by mihailik** [comment]
Not cool why? Are you still hoping a real Adobe dev can stray over here to comfort us with empty promises?

Accept it, we're all alone here, nobody can hear our screams. Can as well fill the thread will ASCII art and Pentagon interrogation transcripts -- still nobody would ever turn up. It's a dead issue, they've just forgotten to close it.

**Comment by mackenza** [comment]
what I see is a medium priority issue open in a repo that has 1300+ open issues, with a viable workaround, on a platform that only 6% of Brackets users care about (and I am one of them, btw, and have been very vocal about Linux support and have Linux on 4 of my 6 computers between work and home).

Yes, we would all like to see the issues we care about fixed first, but the above factors make this issue farther down the triage list than we would like. Adobe has 6 resources working on Brackets afaik. They aren't the only ones who can fix issues. The community could fix this issue and just the triage queue just like that... know anyone who can build stuff on Linux?

**Comment by nethip** [comment]
@mackenza Thanks for chiming in and we are a 6 member team, which takes care of everything relating to Brackets.

@zaggino @MarcelGerber Does it make any sense to spawn an experimental `bracketron` Linux build and distribute that?

@mihailik 
Right now we are very tight on the resourcing and we are trying really hard to complete our existing stories. As mentioned by @mackenza, we have called for help multiple times, but we could not get anyone from the community, you could actively look into this issue. I did spend couple of days looking into issue(the related CEF 2171 migration issue) and realized that needs more time. We are willing to collaborate if there is anyone interested in taking the lead in fixing the issue. Thanks!

**Comment by zaggino** [comment]
I haven't touched `bracketron` for a while now, I'm not sure what Linux users would be missing if we went that way. @mackenza you use Linux right? how does `bracketron` compare to the `brackets-linux`? And I mean what's missing in `bracketron`, not what's better there (native menus and so...)

**Comment by mackenza** [comment]
@zaggino I have not tried Bracketron recently. I feel like it's a waste of time if it's not going to be used by core. We proved it's worth in the PoC but it needs to be completed or dropped. I don't think it's ready for production on Linux or any platform as is. Do you?

**Comment by nethip** [comment]
@mackenza Our primary goal is to unblock folks who are running Brackets on Linux. I was only throwing one of the options that we could exercise. It could be a bad idea too :smile: . 

Of course, there might be some `brackets-shell` specific things(like live developement) that need to be matched. IMHO, if all the workflows are met, we should think about this approach.

About `electron-shell` replacing brackets-shell, the only blocker (which kind of is a big one), is the dark UI not present in `electron-shell`. I did have a look at `electron-shell` and porting our dark UI to electron-shell is not going to be that trivial as `electron-shell` is completely dependent on `chromium` to manage the window unlike `brackets-shell` which does window management.

**Comment by nikaro** [comment]
Why not embed the libraries in Brackets, like Atom does ?

A symbolic link `ln -s /path/to/atom/libgcrypt.so.11 /lib/x86_64-linux-gnu/` make Brackets work.

**Comment by mackenza** [comment]
@nethip I don't think your idea is bad, I am just thinking of the logistics here. Taking Bracketron beyond a PoC to complete all Brackets functionality is an effort. I am not sure @zaggino and @MarcelGerber (who have been the primary contributors to it) are going to be super interested in putting their efforts into completing Bracketron if the final use is only going to be for Linux which is such a small percentage  of users (even less if it's as an experimental build for only those Linux users unwilling to install libgcrypt11).

So while the idea is fine, I think the execution of it would be problematic.

but that is my opinion.

**Comment by ProfitHunter** [comment]
download install libgcrypt11.xx.deb using Debian 7 version deb package
 url:https://packages.debian.org/search?keywords=libgcrypt11
 1.5.0-5+deb7u3 
 i tested,it's working.

**Comment by nikaro** [comment]
@ProfitHunter don't come here to leave a comment with a bad solution without reading what have already been answered.

**Comment by Romane-T** [comment]
Take a good look. This is the solution - install libgcrypt11 from the 
earlier version of Debian - that has been touted from the start when I 
first reported this issue all them months ago, of which this thread is a 
spinoff. In fact, it is about the only solution that is actually 
successful for everybody

So perhaps your criticism is completely unhelpful - clearly you yourself 
have been "without reading what have already been answered."

**Comment by nikaro** [comment]
This is not the solution, this is just a bad workaround.

Already been answered :
https://github.com/adobe/brackets/issues/10255#issuecomment-104398810
https://github.com/adobe/brackets/issues/10255#issuecomment-123216549

**Comment by Romane-T** [comment]
Anyone can criticise, so at least I know your name - Anyone.

If you have a better solution, post it, or, even better, learn how to 
fix it and offer it to the team. THAT would be useful !!! Far more 
useful than one of the umpteen variations of the word "Bullshit".

**Comment by nikaro** [comment]
My suggestion for a better solution have already been posted : https://github.com/adobe/brackets/issues/10255#issuecomment-124482843 (did you read the thread instead of annoying me ?)

Have a nice bullshitting day :wink:

**Comment by abose** [comment]
Guys please maintain the house rules
@nethip @ryanstewart we need to adopt the open code of conduct similar to this.
http://todogroup.org/opencodeofconduct/#Atom/opensource@github.com

**Comment by abose** [comment]
Any suggestion, however minute is welcome in the group, a workaround/ a bad workaround is a workaround none the less. Please keep posting your findings/suggestions in the thread. Help is always welcome here :)

**Comment by nikaro** [comment]
My apologies, my answer to ProfitHunter may have been rude in the form. But I maintain my statement that install the libgcrypt11 deb package is a very bad solution. You install a (may be) outdated and not maintained package with security issues or bugs. And because it is installed manually it will never upgraded. So your system is vulnerable and could be considered as compromise.

A better way for this workaround would be to add in your `sources.list` the Debian (or Ubuntu) repo where this libgcrypt11 package is still maintained, and configure APT pinning to use just this package in this repository. This way would be far more secure and viable.

To get back to my suggestion, embed the library (like Atom does) will avoid OS/distro dependency on this point.

**Comment by nethip** [comment]
@abose that is a good idea. Let us discuss this offline.

@nikaro I would request you to calm down. All the people on the thread know the solution, so please be courteous to people you are trying to help you, even it the proposed solution is wrong. Thanks!

**Comment by nikaro** [comment]
I'm calm, just a little bit annoyed by these already known and already posted "install libgcrypt11.deb" workarounds. I follow this issue to know the progression of the work on this, not to keep receiving "install libgcrypt11.deb" notifications.

And I'm more into [Cocker's rules](http://sl4.org/crocker.html) than OpenCodeOfConduct.

**Comment by nethip** [comment]
@nikaro I have gone ahead and updated your comment from cocker's rule to Crocker's rule. Please take care of typos for especially words like these. This is a very respectable community. Even Crocker's rule clearly says

"Note that Crocker's Rules does not mean you can insult people; it means that other people don't have to worry about whether they are insulting you."

The bottom line is be courteous to folks whether it is Crocker's rule or something else. If you have a different opinion, then we are sorry to say that we will not be able to encourage your participation here.

**Comment by abose** [comment]
Guys, please be nice to each other :)
![image](https://cloud.githubusercontent.com/assets/5336369/8953950/da5e0352-35ff-11e5-8fb7-767e27396041.png)

**Comment by nethip** [comment]
:)

![image](https://cloud.githubusercontent.com/assets/1577184/8954179/75b11ff0-3601-11e5-9f4e-5646c97c2765.png)

**Comment by nikaro** [comment]
@nethip thanks for fixing the typo. But a cocker is just a dog, why would I be especially vigilant for this word ? Did I insult someone in my comments ? I just have been rude to @ProfitHunter, and I apologized as soon as @abose call us to order. What do you want more ?

Crocker's rules put aside. Please note that not everybody is fluent in english and have the vocabulary to put these "courteous" (<-- you make me discovered this one) words and make nice sentences.

![sad cocker](http://www.aciclico.com/wp-content/uploads/2009/11/cocker.jpg)

**Comment by nethip** [comment]
@nikaro Sorry if I sounded hurting/rude. That was not my intention. I just want to make sure we have a friendly community and that everyone is respected. I am going to stop with this :+1:

**Comment by Romane-T** [comment]
Yes, I do recall your suggestion, but also recall the reasons it has not 
yet been taken up.

Just roll your eyes and delete the messages that annoy you about 
libgcrypt11. Going to be a few more of them before this issue is 
resolved by the Brackets team.

Those issues you raise don't bother all that many, who will and are 
quite happily (and without problems) running a version of libgcrypt11 
from pre-jessie days; even if they are aware of the issues, most don't 
care or have the interest. The reality is that it works and is simple, 
and that is all most are interested in. So far as apt-pinning - another 
reality is that not too many interested in going through the effort, 
myself included. And until the team do put in place a better solution, 
just being able to get Brackets running will be all that is needed to 
make most happy.

Just for those interested, running the 1.4 Windows version under Wine 
fails with an exception. Of course, a virtual Windows machine will not 
have this problem :)

**Comment by mihailik** [comment]
The biggest problem on this thread is that not one person explained **what is the reason for this bug, what is the workaround, and what is the downside of the workaround.**

Giving a link to some weird page on Internet, and hand-waiving *'frob that blick but not schmob that driubit'* is next to nothing. Normal people don't know your lingo, and aren't into intimate relationship with deb/conf/sudo sex toys.

Of course Adobe developers are real culprits here. You guys have time to make ridiculously dismissive statements, but no time to clarify the situation?

Suggesting someone steps up and fixes the bug requires **you, Adobe** making it clear what the bug is, why it happens, and ideally outlining a plan to fix it.

Or not. Keep it muddled, that works for me. Much fun looking at how public gets really annoyed and hurts your public image by something that I suspect is a minute issue. Good luck!

**Comment by nikaro** [comment]
@mihailik I'm not an Adobe developer, but here is a short answer from what I have understood :

- **reason** : Brackets use a CEF version which depend on libgcrypt11, up-to-date distros ship libgcrypt20 instead of libgcrypt11
- **ugly and dangerous workaround** : grab libgcrypt11 from somewhere on the Internet and ~~break~~ ~~compromise~~ install it on your system
- **still ugly but less dangerous workaround** : add oldstable repo in your `sources.list` and configure APT pinning to get libgcrypt11 from it

So to fix this issue, it seems that they need to upgrade their CEF version and dependencies of the packages. A quick fix waiting for CEF upgrade could be to embed the libgcrypt11 library.

**Comment by mihailik** [comment]
Thanks @nikaro.

I'm afraid it still doesn't make much sense. Well, the first workaround I've tried ages ago, but Ubuntu refused to install that pesky lib11.

The second one would be great for a blockbuster movie: convincing techy zing, yet nobody in the audience would work out if it's a fake jargonism.

**Comment by mackenza** [comment]
@mihailik your sarcastic responses to this thread are not welcome. Please observe some level of community respect and stop your antagonistic replies. Thank you in advance.

**Comment by mihailik** [comment]
@mackenza please don't let my manners stopping you from addressing the underlying issue.

It'll soon be a year since the issue reported, so I am sure you're eager to break the silence now.

**Comment by mackenza** [comment]
If by "soon be a year" you mean it 5 months from now, you are bang on.

**Comment by Romane-T** [comment]
Just an observation or three. Nothing more. nothing less.

1. This package is provided by Adobe for free. Now, if one were paying 
for it, then these complaints would be most valid. But it is literally 
being given away. Reminds me of the saying about looking a gift horse in 
the mouth.
2.  If one has sufficient distaste for what is being given free and the 
work-around being used to get it to run at this time, there is always 
another editor. Bluefish comes to mind, a likewise excellent editor.
3. There is nothing to prevent one from getting the source, patching it 
how one wants and re-compiling it, and one is not even obligated to do 
any more with it than own personal use. And if one does not wish to go 
through that process and effort, then see point 2.
4. The issue is only 5 months old. Looking at other bugs in the distros 
shows some bugs waaaay more than that. Without the degree of complaint I 
have seen over the last couple of days here, and some of those are much 
more critical than this issue.
5. The team developing Brackets is small. As explained by one member in 
an earlier post, the Linux users are only a tiny portion of the user 
base. Yes, it would be nice to be at the top of the user list, but the 
reality is that Linux users are not.
6. The greater majority of the Linux users have no interest in any more 
than having it working. The reality is that it is only a tiny tiny 
proportion of Linux users making an issue out of libgcrypt11. That 
silent majority is an implicit vote of confidence in the Brackets team.
7. If this libgcrypt11 issue really bothered the few users who are on 
Linux, then the Linux version would quickly find itself without any 
Linux users. Refer to point 6. Then, there would be zero encouragement 
to fix any Linux issues, if there were even a Linux version put out
8. I find it great that the issue is still being kept,in the forefront. 
It would be much more pleasant for everyone if the postings could be 
kept cordial. Remember, the butt that you kick today may be the butt 
that you need to kiss tomorrow.

**Comment by mackenza** [comment]
@Romane-T wins this thread... he is exactly bang on 1-8

**Comment by mihailik** [comment]
'5 months' stretching February — July, 'only tiny minority' including every single one on the latest Ubuntu, the issue 'being kept on the forefront' of what exactly?

I can't imagine writing a concise explanation in layman's terms is a task worthy all this bickering.

**Comment by dwilmer** [comment]
The work on including the new CEF, which (among others) removes the dependency on libgcrypt11, is tracked in issue #11047. They want to include the new version, but there are still some problems before doing so. The aforementioned issue tracks and details those problems, so you can understand what is going on and, if you have the time and skills needed, contribute.

**Comment by ryanpcmcquen** [comment]
For anyone running into this issue on Slackware-current, I have a build for the old libgcrypt that will ONLY install the necessary libraries, and coexist with the newer libgcrypt.

https://github.com/ryanpcmcquen/ryanpc-slackbuilds/tree/master/unofficial/libgcrypt15

**Comment by scottie** [comment]
So im confused seems fix's are all posted why has this not been updated into the main code ?

**Comment by citizen6** [comment]
Brackets will not run by altering the dependency in the installation files to work with libgcrypt20. 

You'll need to manually install libgcrypt11 for your architecture from the debian packages. Alternatively, add the wheezy repository in your sources.list file and apt-get install -f will solve the dependency problem. 

Tested on Debian Jessie i386 and Stretch amd64.

**Comment by fizzvr** [comment]
@citizen6 manually installed from wheezy :beers:

**Comment by doug65536** [comment]
Continuing to claim support for Linux is pretty much a lie now. It doesn't take a year to fix one dependency version.

**Comment by abose** [comment]
The Linux builds of brackets have never graduated from `experimental` stage. If you go to `help>about brackets` it is clearly stated that Linux builds are experimental. 

Free and open source software is when people come together to build awesome software like Brackets. As Brackets is an open source software built by community members, support for a specific system will only be as good as the will of the community interested in contributing to it. Issues like this in windows or mac never stay open for long like this - thanks in large part to the community. 

We request the support of Linux community members in ironing out this issue.

**Comment by brbsix** [comment]
Just in case anyone is curious, the solution to this (as an end-user) is very simple. It requires you to have download or extracted *libgcrypt.so.11* from elsewhere beforehand. I'll post my quick hack for resolving this issue. I'll be making use of `debtool`, my own homegrown tool to unpack and repack *Brackets.1.5.Extract.64-bit.deb*, but you can use `dpkg` primitives instead if you're familiar with them. You can download `debtool` here: https://github.com/brbsix/debtool

1. Unpack *Brackets.1.5.Extract.64-bit.deb*

        debtool -u Brackets.1.5.Extract.64-bit.deb

2. Remove the package's dependence on `libgcrypt11`

        sed -i 's/ libgcrypt11 (>= [0-9.]\+),//' ./brackets_1.5.0-16538_amd64/DEBIAN/control

3. Copy *libgcrypt.so.11* into *./opt/brackets/*.

        cp ~/Downloads/libgcrypt.so.11 ../brackets_1.5.0-16538_amd64/opt/brackets/

4. Rebuild the package

        debtool -b ./brackets_1.5.0-16538_amd64

5. Install *brackets_1.5.0-16538_amd64.deb*

**Comment by tharmann** [comment]
brbsix - brilliant, plain and simple.  I just tested on a fresh install of jessie and it worked a treat.  Thanks for the debtool as well ;)

**Comment by abose** [comment]
Thanks @brbsix for suggesting the fix. will checkout if a modified installer can be tested and posted for the next release.

**Comment by abose** [comment]
In the mean time, if anyone is interested in changing brackets installer build scripts for Linux, please see https://github.com/adobe/brackets-shell and https://github.com/adobe/brackets/wiki/Building-Brackets-Releases

**Comment by navjotjsingh** [comment]
I tried the method posted by @brbsix on Ubuntu 15.10. Installation went fine but I can't seem to run Brackets. I have tried launching it multiple times, but nothing happens.

**Comment by brbsix** [comment]
@navjotjsingh Can you confirm */opt/brackets/libgcrypt.so.11* exists? If it does exist and you're still having an issue, I guess I'd be curious as to what release of Brackets you're using (e.g. 32bit or 64bit, with Extract or not).

I hate to recommend downloading debs from third-party sources, but I've repacked all the v1.5 releases and included links at the bottom of this page: http://brbsix.github.io/2015/10/28/fix-missing-libgcrypt11-for-brackets-text-editor/

**Comment by navjotjsingh** [comment]
Yes it does. I am using 64 bit Brackets [with extract].

**Comment by navjotjsingh** [comment]
The deb I downloaded from your site worked fine. Thanks.

**Comment by rieftux** [comment]
@brbsix Thanks, that's work for me on Debian 8 64bit.

**Comment by lucaszanella** [comment]
i didn't read any of this but can you guys make brackets work in debian 8 please?????? for the official deb package

thanks bye

**Comment by jabreul** [comment]
The approach that @a-moses, @sarathms are following is nice. But I tried installing the dependency from the debian repo, the version from wheezy, even thought I'm using Jessie, from https://packages.debian.org/wheezy/libgcrypt11

Since all the dependencies are met, it will install without problem and Brackets will use the library that is expecting. The installation will be completed without any problem and matching the requirements.

**Comment by brbsix** [comment]
@abose I looked into updating the build scripts and have submitted a fix at https://github.com/adobe/brackets-shell/pull/541. I wasn't sure what branch to submit the PR to, so I just did it against master. I guess alternatively the library could have been bundled into the CEF itself or something, but this was the most straightforward solution.

**Comment by abose** [comment]
@brbsix Thanks very much for the pr :) 
Will verify it and reply back.

**Comment by pjschaafsma** [comment]
I just cannot believe such a core issue has been around for over a year, and still not dealt with!!!

This says something about the developers behind this software. I'm skipping Brackets. Probably forever.

**Comment by GogoAkira** [comment]
pjschaafsma, but that's why brbsix fixed it for us on his page, hehe

**Comment by ldez** [comment]
New version (1.6) is out but no official correction available.

**Comment by GogoAkira** [comment]
@brbsix , brbsix we need you.  Please do the same thing with the new version, add this library, because some of us get lost trying to do that.

**Comment by brbsix** [comment]
@GogoAkira No problem. Links to the latest fixed packages can now be found at the bottom of [this page](https://brbsix.github.io/2015/10/28/fix-missing-libgcrypt11-for-brackets-text-editor/) or here:

**Release 1.6 + Extract bundle**

* [Brackets.1.6.Extract.64-bit.deb](https://www.dropbox.com/s/2k5ym33v9qchin1/Brackets.1.6.Extract.64-bit.deb?dl=1)
* [Brackets.1.6.Extract.32-bit.deb](https://www.dropbox.com/s/4gwp4elj6f99m5j/Brackets.1.6.Extract.32-bit.deb?dl=1)

**Release 1.6**

* [Brackets.Release.1.6.64-bit.deb](https://www.dropbox.com/s/6gllagx95tm1xmp/Brackets.Release.1.6.64-bit.deb?dl=1)
* [Brackets.Release.1.6.32-bit.deb](https://www.dropbox.com/s/ortzy9aqkw02tb4/Brackets.Release.1.6.32-bit.deb?dl=1)

I also submitted a pull request to fix this the correct way. I wonder why it didn't make it into the new release...

**Comment by GogoAkira** [comment]
heey, thank you, awesome, i'm getting it right now, at first i did try to rebuild it myself according to your instrucions, heh but i'm not yet at that level, i'll get to that later, so thank you.  Yeah, I wonder what os do they use themselves when building brackets, because it probably works for them heh.

**Comment by dvorapa** [comment]
@brbsix :thumbsup:

**Comment by antivanov** [comment]
Installing libgcrypt11 worked for Debian 8.3.0 Jessie and Brackets now starts up:

1. `wget ftp://ftp.debian.org/debian/pool/main/libg/libgcrypt11/libgcrypt11_1.5.0-5+deb7u3_amd64.deb`
2. `sudo dpkg -i libgcrypt11_1.5.0-5+deb7u3_amd64.deb`
3. `sudo dpkg -i Brackets.Release.1.6.64-bit.deb`

**Comment by lucaszanella** [comment]
too much text, too much time, new versions appeared, and no fixes

I'm leaving brackets

bye

**Comment by dvorapa** [comment]
It looks like linux isn't important for Brackets' devs (this issue or years old issue of popular distros' own navbar instead of that HTML one or another years old issue of Run with option not working with Brackets on Linux). But what instead? I'd like to move on, but I'd like almost all of Brackets features and Brackets look.

**Comment by brbsix** [comment]
@dvorapa [Atom](https://atom.io/) is also built on the CEF (Chromium Embedded Framework) and has a similar look and feel. It is open source as well, and has the backing of GitHub.

If you're willing to consider proprietary software, I think [Sublime Text 3](https://sublimetext.com/) is the very best graphical text editor available. It has the feature set of an IDE (especially when you consider all the plugins available) but is native and blazing fast... Also it is being actively developed and is well-supported.

**Comment by dvorapa** [comment]
@brbsix I know both, but both are not looking so good. And I am esthetic person, which must code in something beautiful for eye (and also which contains some other stuff like plugins, replace in the whole project, connection between functions in different files, czech localization etc.). Do you know if any of these two alternatives have czech localization?

**Comment by brbsix** [comment]
In light of the news that Adobe is [unable to bundle](https://github.com/adobe/brackets-shell/pull/541) *libgcrypt.so.11* as virtually all other projects using CEF have done (e.g. Atom text editor, Nylas email, OpenBazaar, and Spotify clients) I want to propose a new solution. [AppImage](http://appimage.org/) is a new type of package format that exemplifies a "package once, run everywhere" approach. It runs on nearly any Linux distro, requires no installation, and has no system dependencies. After speaking with it's creator, the package now includes *libgcrypt.so.11* and should remain compatible with any Linux distro in perpetuity, irrespective of any future deprecations or requirements added by Adobe.

The Brackets AppImage package is available [here](https://bintray.com/probono/AppImages/Brackets/). To download and run Brackets v1.6.0.16680:

    # download the AppImage (substitute whatever command you use to download things)
    curl -L --remote-name-all https://bintray.com/artifact/download/probono/AppImages/Brackets-1.6.0.16680-x86_64.AppImage

    # mark the file as executable
    chmod u+x Brackets-1.6.0.16680-x86_64.AppImage

    # run it
    ./Brackets-1.6.0.16680-x86_64.AppImage

**Comment by lennonjesus** [comment]
AppImage format is very cool!

**Comment by Romane-T** [comment]
I am impressed. Just fired it up briefly, and can only say - I am 
impressed. Very impressed.

The link given below (/The Brackets AppImage package is available //here 
<https://bintray.com/probono/AppImages/Brackets/>/) is a little 
misleading, as it eventually leads to dl.bintray... rather than just 
bintray...., and thus gives a 404 error. Found that the better link for 
the "here" given in your post is https://bintray.com/probono/AppImages, 
and that also shows the user what else is available.

A suggestion to the Brackets devs. Not only does this get around the 
libgcrypt11 issue, it also appears to allow it to run in more 
distributions *without* relying on other distros than Debian and its 
derivatives, to build and distribute. My experience in playing with 
different distro's is that there is a strong degree of inconsistency and 
often old versions being supplied. Considering that so much is the work 
of volunteers, every effort must be given the accolade of encouragement. 
But, this AppImage format can only work in favour of Brackets, and 
reduce the amateurish image of the project which is starting to appear 
both through this inconsistency and through this pitiful inability to 
move to the version of libgcrypt being carried now by most distros.

Lets face it - the Linux community will only tolerate matters so far, 
and then you lose some of the user base. Perhaps Adobe don't care about 
their Linux users, but disregard any segment of your user base and it 
can only have negative ramifications over the longer term. Windoze has 
been losing to both Apple and Linux progressively for a while, so it can 
only be a matter of time. And I strongly doubt that I am alone in saying 
"why should I run something under Wine when the developer can't be 
bothered catering to the Linux user base?"

Yes, I have tried Atom (thank you to the person who posted with that 
alternative), and should it come up to my needs and requirements, then 
.... well, loyalty has to be earned, and so far, Atom doesn't quite suit 
my needs, but.... Again, I doubt strongly that I am alone in these 
sentiments.

Brackets deserves better than it is being given. If Adobe can't do it, 
Brackets will eventually fade into a pipe-dream in the sky. And that 
would indeed be a shame, for it is, hands down (in my personal opinion) 
the best editor available - for the nonce. libgcrypt11 has been obsolete 
for so long now, that I fully understand those who have already walked. 
And aren't legal issues open to negotiation, most especially in an 
open-source continuum?

Come on Adobe, stop dragging your heels. Support your Brackets 
developers, every one of whom will be doing their best, by putting in 
more effort than it takes to shrug and thus letting them badly down.

Romane

**Comment by mevsme** [comment]
Can not install any version at Ubuntu 15.04, Application center shows libgcrypt11 requiered >:-|

**Comment by AlterEgo7** [comment]
http://www.webupd8.org/2015/04/fix-missing-libgcrypt11-causing-spotify.html

**Comment by probonopd** [comment]
@mevsme 
> Can not install any version at Ubuntu 15.04, Application center shows libgcrypt11 requiered >:-|
Does https://bintray.com/probono/AppImages/Brackets/_latestVersion#files work for you? Just download the AppImage, chmod a+x, and run.

**Comment by ZiTAL** [comment]
you can install libcrypt11 from here:

http://security.debian.org/debian-security/pool/updates/main/libg/libgcrypt11/libgcrypt11_1.5.0-5+deb7u4_amd64.deb

**Comment by dvorapa** [comment]
@brbsix thanks for giving me ideas for another code editors. I finally moved out from Brackets (because of poor support for Linux, currently I even can not install three of latest versions on newest fully updated 3rd most used Linux distribution in the world and there also is not any support for native OS menus, I can not open any file by clicking on it in native OS file manager - opens plain Brackets and not file - and a stability of Brackets is really bad these days. Some bugs are also there for more than three versions - e.g. the annoying notification after first install, which can not be closed by clicking the x button, the only way is to restart Brackets). The only advantage of Brackets at the moment is its localization (I still translated every version by PR, but better than nothing)

**Comment by alemonteiro** [comment]
As @jws121295  said on https://github.com/adobe/brackets/issues/10465,

adding wheezy to sources.list works great.

**Comment by thoroc** [comment]
thanks @ZiTAL, that worked like a charm on Ubuntu 15.04 :)

**Comment by dinusuresh** [comment]
Faced the same issue on a fresh install of Kubuntu (KDE on Ubuntu 16.04 LTS).

Installing libgcrypt11 fixed it!

For ubuntu 16.04, install libgcrypt11 from here:
32-bit: https://launchpad.net/ubuntu/+archive/primary/+files/libgcrypt11_1.5.3-2ubuntu4.2_i386.deb
64-bit: https://launchpad.net/ubuntu/+archive/primary/+files/libgcrypt11_1.5.3-2ubuntu4.2_amd64.deb

**Comment by Rafael1981** [comment]
thanks Zital, it worked right away

**Comment by rmalheiros** [comment]
It was simple to resolve on  Debian Jessie

Just copy /etc/apt/sources.list -> /etc/apt/sources.list.d/oldstable.sources.list
Leave just those lines that begin with deb and deb-src.
For me, it was:
`deb http://ftp.br.debian.org/debian/ oldstable main`
`deb-src http://ftp.br.debian.org/debian/ oldstable main`

then run as root
`apt-get update`
`dpkg -i <Brackets File>`
`apt-get -f install`
`brackets`

For Red Hat distros, use
`yum` or `dnf` instead of `apt-get`

**Comment by ldez** [comment]
@jasonsanjose  @abose @nethip Can we expect a properly fix?
The issue with `libgcrypt11` is critical on Debian/Ubuntu.

![cat](http://i147.photobucket.com/albums/r289/Wootsee/Shrek%203/Chat_potte4.png)

**Comment by piesha** [comment]
+1 from someone negatively impacted by this issue.

**Comment by Romane-T** [comment]
+1 from another impacted

**Comment by nethip** [comment]
@ldez Unfortunately we don't have adequate bandwidth to look at shell side of things. We were hoping to get some help from the community but we couldn't get any :(.

**Comment by ficristo** [comment]
@nethip I'm (slowing) looking at updating the shell to match the cefclient example in the hope to help with this.
I need someone to review my PRs...

**Comment by ficristo** [comment]
I don't think is fair to say the community isn't helping: there are already some PR on the shell.
(They are not specific of this issue but still)
https://github.com/adobe/brackets-shell/pull/351
https://github.com/adobe/brackets-shell/pull/425
https://github.com/adobe/brackets-shell/pull/511
https://github.com/adobe/brackets-shell/pull/518
If you look at the first PR was opened in 10 Oct 2013, IMO is that is blocking contributors...

**Comment by ldez** [comment]
@nethip The PR to fix the problem is here (I hope): https://github.com/adobe/brackets-shell/pull/501

Before this PR, @brbsix have made another PR https://github.com/adobe/brackets-shell/pull/541

**Comment by nethip** [comment]
@ficristo I was talking particularly about this issue itself. We are really appreciative of all the work the community is putting in to make this a great product. I respect that :+1: . 

About your PRs, sorry that we are not able to have a look at your PRs. I can understand the frustration. But we are really squeezed between the upcoming releases of Dreamweaver Beta and Brackets. We are doing our best to be able to review all pending PRs(we are infact putting in extra efforts). Your PRs are my `to-review` list. I will find some time to review your PRs. Really appreciate all your efforts.

@ldez We were really hopeful about https://github.com/adobe/brackets-shell/pull/541 but it got turned down because of the legal issues. However https://github.com/adobe/brackets-shell/pull/501 looks promising. This is not tested yet. Will you help us in validating this fix?

**Comment by nethip** [comment]
Please refer to [this comment](https://github.com/adobe/brackets-shell/pull/501#issuecomment-225966380) by @brbsix . Seems like we are out of luck again.

**Comment by nethip** [comment]
@ficristo I am done reviewing your PRs that you had raised in the `brackets-shell` repo. Please address the code review comments.

**Comment by ficristo** [comment]
@nethip Thank you for the reviews!

**Comment by ficristo** [comment]
For whoever wants to help, https://github.com/adobe/brackets/issues/12640 is related.

**Comment by TimHandy** [comment]
+1 from another impacted. Linux Mint 18.

**Comment by rodya-mirov** [comment]
Can't install either (Ubuntu). This issue is 2 years old!

**Comment by ficristo** [comment]
Given the current situation on Linux, and my doubts that anything will change anytime soon, I would like to suggest to use [brackets-electron](https://github.com/zaggino/brackets-electron).
Basically it's Brackets but with Electron as the underline shell. I think it's worth a try.
Here a release: https://github.com/zaggino/brackets-electron/releases/tag/v1.8.1

Last year, electron was bring up as an alternative of our shell. @zaggino and some other put together a proof of concept, but the lack of a native dark theme was a blocker to continue the integration.
Lately @zaggino has put some effort to make brackets-electron more stable.

If you have more questions on this I suggest to ask them on Slack in the brackets-electron channel or on
[brackets-electron issues](https://github.com/zaggino/brackets-electron/issues) or open a new issue here to discuss only this.

A __warn__: AFAIK there isn't any plan to switch to it.

We know the problem but there aren't many users on Linux to justify an effort here. Or at least, this what I hear everytime about Linux situation and personally I think is really bad. I would like to help more but I can't do much :sob: 

I lock this issue: more comments here will not increase the priority but only add noise.

**Comment by saurabh95** [comment]
Fixed By: https://github.com/adobe/brackets-shell/pull/619

**Comment by noxxex** [comment]
try apt --fix broken install

**Comment by GogoAkira** [comment]
Are we still here guys, it's 2019 now :)
